libsolidity/analysis/SyntaxChecker.cpp

```diff
@@ -335,7 +334,11 @@ bool SyntaxChecker::visit(FunctionTypeName const& _node)
 bool SyntaxChecker::visit(VariableDeclarationStatement const& _statement)
 {
 	// Report if none of the variable components in the tuple have a name (only possible via deprecated "var")
-	if (boost::algorithm::all_of_equal(_statement.declarations(), nullptr))
+	if (std::all_of(
+		_statement.declarations().begin(),
+		_statement.declarations().end(),
```

Comment:
I'm pretty sure that @chriseth particularly doesn't like the STL versions of these that require ``begin()`` and ``end()`` iterators...

But to avoid that we'd need ``std::ranges::all_of`` in C++20 (or ranges-v3 :-) ).

Reply:
Make sure to link to this on the ranges-v3 discussion.

Reply:
I think there's no dedicated issue justifying ranges-v3 in general, just https://github.com/ethereum/solidity/issues/8860 (which is hereby linked).